client/student/app/lecture/comment.panel.directive.js

```diff
@@ -24,11 +24,34 @@
             console.log('Ready (Comment Panel Link)');
         }
     }
-    
     /* @ngInject */
-    function CommentPanelController() {
+    function CommentPanelController(lectureFactory) {
         console.log('Ready (Comment Panel Controller)');
-        
-        var commentPanel = this;       
+       
+        var commentPanel = this;  
+        commentPanel.upVoteComment = upVoteComment;
+        commentPanel.downVoteComment = downVoteComment;
+
+        function upVoteComment(comment){
+            console.log('+1');
+            lectureFactory.upVoteComment(comment.id, 1, function(respone){
+                console.log('Success');
+                comment.rating = 1;
+
+            }, function() {
+                console.log('error');
+            });
+        } 
+
+        function downVoteComment(comment){
+            console.log('-1');
+            lectureFactory.upVoteComment(comment.id, -1, function(respone){
```

Comment:
The name upVoteComment in lectureFactory is a bit confusing since you pass the vote as a parameter. It should be renamed to 'setCommentVote' or something like that.